configure.ac

```diff
@@ -136,43 +136,58 @@ AC_CHECK_HEADER(valgrind/memcheck.h, [],
 fi
 
 AC_CACHE_CHECK(whether ld accepts --version-script, ac_cv_version_script,
-    if test -n "`$LD --help < /dev/null 2>/dev/null | grep version-script`"; then
+    [if test -n "`$LD --help < /dev/null 2>/dev/null | grep version-script`"; then
         ac_cv_version_script=yes
     else
         ac_cv_version_script=no
-    fi)
+    fi])
 
 AM_CONDITIONAL(HAVE_LD_VERSION_SCRIPT, test "$ac_cv_version_script" = "yes")
 
 dnl Disable symbol versioning with icc + ipo, with it ipo is disabled by icc.
 dnl The gcc equivalent ipo (-fwhole-program) seems to work fine.
 AS_IF([case "$CFLAGS" in
-         *-ipo*) true ;;
-         *)      false ;;
-        esac],
-      [AC_MSG_NOTICE([disabling symbol versioning support with -ipo CFLAG])],
-[
-
-AC_CACHE_CHECK(for .symver assembler support, ac_cv_asm_symver_support,
-	[AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[]],
-		[[asm("symbol:\n.symver symbol, api@ABI\n");]])],
-		[ac_cv_asm_symver_support=yes],
-		[ac_cv_asm_symver_support=no])])
-
-]) dnl AS_IF
-
-if test x$ac_cv_asm_symver_support = xyes; then
-	AC_DEFINE([HAVE_SYMVER_SUPPORT], 1, [assembler has .symver support])
-fi
+	*-ipo*) true ;;
+	*)      false ;;
+	esac],
+	[
+		AC_MSG_NOTICE([disabling symbol versioning support with -ipo CFLAG])
+		ac_icc_hack=0
```

Comment:
I'm assuming that this ac_icc_hack is something wonky for the icc compiler?  It should probably not begin with "ac_" (I think that's supposed to be reserved for autoconf stuff), and there should probably be a comment explaining what it's for/why you need to disable .symver for icc in some cases.